hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/service/ContractCallService.java

```diff
@@ -67,14 +76,32 @@ private HederaEvmTransactionProcessingResult doProcessCall(final CallServicePara
                             body.getCallData(),
                             body.isStatic());
 
-            counter.increment(txnResult.getGasUsed());
+            Counter gasPerSecondCounter;
+            if(!txnResult.isSuccessful()) {
+                gasPerSecondCounter = gasPerSecondMetricMap.get(CallType.ERROR);
+                gasPerSecondCounter.increment(txnResult.getGasUsed());
```

Comment:
Can move to a more generic listener method to stay DRY:

```suggestion
            if (!txnResult.isSuccessful()) {
                onCompleted(body, txnResult);
```

```java
private void onCompleted(CallServiceParameters body, HederaEvmTransactionProcessingResult result) {
    var counter = gasPerSecondMetricMap.get(body.getCallType());
    counter.increment(result.getGasUsed());
}
```